bugsnag-android-core/src/androidTest/java/com/bugsnag/android/ObserverInterfaceTest.java

```diff
@@ -50,59 +50,42 @@ public void tearDown() {
     @Test
     public void testAddMetadataToClientSendsMessage() {
         client.addMetadata("foo", "bar", "baz");
-        List<Object> metadataItem = (List<Object>)findMessageInQueue(
-                NativeInterface.MessageType.ADD_METADATA, List.class);
-        assertEquals(3, metadataItem.size());
-        assertEquals("foo", metadataItem.get(0));
-        assertEquals("bar", metadataItem.get(1));
-        assertEquals("baz", metadataItem.get(2));
+        StateEvent.AddMetadata msg = findMessageInQueue(StateEvent.AddMetadata.class);
+        assertEquals("foo", msg.getSection());
+        assertEquals("bar", msg.getKey());
+        assertEquals("baz", msg.getValue());
     }
 
     @Test
     public void testAddNullMetadataToClientSendsMessage() {
```

Comment:
This seems to be a duplicate of `testAddNullMetadataToMetadataSendsMessage`. We're now testing everything through `Client`, so it seems we can remove the `ToClient`, `FromClient` and `ToMetadata` part of the test names and remove this test.

Reply:
I have removed this test, renamed the other test cases, and removed all duplicates.